File: CoreApiApp/WebApp/Data/Entities/Department.cs
namespace CoreApiApp.Data.Entities;

public class Department
{
    public int Id { get; set; }
    public Guid Guid { get; set; }
    public required string Name { get; set; }
    public int HospitalId { get; set; }
    public Hospital Hospital { get; set; }

    public string CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
    public ICollection<Staff> Staffs { get; set; } = null!;
    public ICollection<Schedule> Schedules { get; set; }


}

File: CoreApiAppPython/models/department.py
from sqlalchemy import Column, Integer, String, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from models.base import Base

class Department(Base):
    __tablename__ = "department"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String, index=True)
    hospital_id = Column(Integer, ForeignKey("hospital.id"))
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), onupdate=func.now())

    hospital = relationship("Hospital", back_populates="departments")
    staff = relationship("Staff", back_populates="department")